gn3/computations/correlations.py

```diff
@@ -422,3 +423,60 @@ def fast_compute_tissue_correlation(primary_tissue_dict: dict,
     return sorted(
         tissues_results,
         key=lambda trait_name: -abs(list(trait_name.values())[0]["tissue_corr"]))
+
+
+def compute_correlation_2(corr_inputs):
+
+    (this_trait_samples, target_trait) = corr_inputs
+
+    trait_name = target_trait.get("trait_id")
+    target_trait_data = target_trait["trait_sample_data"]
+
+    try:
+        (x_vals, y_vals) = list(zip(*list(filter_shared_sample_keys(
+            this_trait_samples, target_trait_data))))
+
+        x_vals = np.array(x_vals, dtype=float)
+        y_vals = np.array(y_vals, dtype=float)
+
+        if len(x_vals) > 5:
+            # remove nan values
+            nans_values = np.logical_or(np.isnan(x_vals), np.isnan(y_vals))
+
+            (corr_coeff, p_val) = scipy.stats.pearsonr(
+                x_vals[~nans_values], y_vals[~nans_values])
+
+            # print(corr_coeff, p_val)
+            return{trait_name: {
+                "corr_coefficient": corr_coeff,
+                "p_value": p_val,
+                "num_overlap": len(x_vals)
+            }}
+
+    except ValueError:
+        return
+
+
+def mp_calculate(this_trait, dataset, n_jobs: int = -1, chunk_size: int = 500):
+    """corr mp reimplementation
+    credit:https://github.com/bukson/nancorrmp
+
+    """
+    this_trait_samples = this_trait["trait_sample_data"]
```

Comment:
*nitpick:* You should really work on how you name your variables.  Stop inheriting this naming scheme from gn2.  Try to use variables that are easy to understand when scheming code.